TeachingRecordSystem/tests/TeachingRecordSystem.TestCommon/TestData.ProfessionalStatusEventBuilder.cs

```diff
@@ -0,0 +1,336 @@
+﻿using TeachingRecordSystem.Core.DataStore.Postgres.Models;
+using TeachingRecordSystem.Core.Events.Models;
+
+namespace TeachingRecordSystem.TestCommon;
+
+public partial class TestData
+{
+    public Task<EventBase> CreateProfessionalStatusCreatedEventAsync(Action<CreateProfessionalStatusEventBuilder>? configure = null)
+    {
+        var builder = new CreateProfessionalStatusEventBuilder();
+        configure?.Invoke(builder);
+        return builder.ExecuteAsync(this);
+    }
+
+    public class CreateProfessionalStatusEventBuilder()
```

Comment:
This should be done as part of `ExecuteAsync` on `CreatePersonProfessionalStatusBuilder` rather than being its own thing (see Alerts for some prior art).

Reply:
- So that we always get a `ProfessionalStatusCreatedEvent` whenever we do `TestData.CreatePerson(p => p.WithProfessionalStatus(...))`
- It's consistent to what we're doing everywhere else.